bigquery/bigquery_create_object_table/main.tf

```diff
@@ -0,0 +1,74 @@
+/**
+* Copyright 2023 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+# [START bigquery_create_object_table]
+resource "google_bigquery_connection" "default" {
+  connection_id = "my-connection-id"
+  location      = "US"
+  cloud_resource {}
+}
+
+data "google_project" "project" {}
+
+resource "google_project_iam_member" "default" {
+  role    = "roles/storage.objectViewer"
+  project = data.google_project.project.project_id
+  member  = "serviceAccount:${google_bigquery_connection.default.cloud_resource[0].service_account_id}"
+}
+
+resource "google_bigquery_dataset" "default" {
+  dataset_id = "my_dataset_id"
+}
+
+resource "google_storage_bucket" "default" {
```

Comment:
Since we're creating a GCS bucket, include a random suffix to ensure uniqueness of the name. For example: https://github.com/terraform-google-modules/terraform-docs-samples/blob/c0e42baa72e0961543d31fa6fcfc8de8fb4cf3b6/vertex_ai/index/main.tf#L19

Reply:
done. added random suffix